### ISSUE TITLE
Supporte scalable resolver

### ISSUE THREAD

**Issue by p53** [privacyidea/privacyidea#1135]
It would be nice to have resolver which is more scalable than SQL databases, e.g. Elasticsearch/Cassandra/Mongodb

**Comment by plettich** [comment]
It should be fairly easy to write a user resolver for any NoSQL database.
The base class it should derive from is the [UserIdResolver](https://github.com/privacyidea/privacyidea/blob/5156a46e94b9b7c4a413719e7c887756a4ccf657/privacyidea/lib/resolvers/UserIdResolver.py#L54).
Is there any _convention_ on how user data is stored in NoSQL databases?

**Comment by cornelinux** [comment]
Till now we did not hear of any such request.
I am not sure if users is the data that is usually strored in mondodb.
To my understanding elasticsearch is not a data source but rather a search engine based on a nosql format.

I would be helpful, if you would provide more information of your idea.

**Comment by p53** [comment]
yup, that's correct, most used user store is LDAP, which is quite bad, it is old, it is not really scalable, it is rigid and not easy to setup, customize...and that's why also commercial LDAP with point and click (not really devops and not really solution to all this problems) is so widespread...Also SQL is not scalable, write scaling with SQL is really pain and setting up and maintain replication also...i would like to have several user stores replicates - e.g. 5 or more, i would like to have easy maintain, create new replica, move data etc...(best would be if also PI would use more scalable backend, i know that's different topic and that would need great amount of work...)

**Comment by NuvandaPV** [comment]
We read our users from existing datastores. Compatibility with legacy products is more important for this product than supporting each new fancy technology. We have yet to encounter IAM-solutions that actually store their users in mongodb or other noSQL-databases.

**Comment by plettich** [comment]
If the userstore provides an HTTP API You can use the the HTTP-Resolver (#2081) which is available since [v3.4](https://github.com/privacyidea/privacyidea/releases/tag/v3.4).
